### PR TITLE
feat: make docs reflecting latest nullability changes.

### PR DESCRIPTION
**Pull request by Lodin**
* Collection type nullability is supported
* Endpoint function parameters are non-omittable

### CODE REVIEW

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 1]
```diff
@@ -92,35 +93,66 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
 * `org.eclipse.jdt.annotation.NonNull`
 * any other annotation (including custom) that has the `nonnull` (case-insensitive) name.
 
-.Required parameters in Java endpoint
+==== Endpoint Functions
+
+For the endpoint function, there are two things that can be nullable:
+
+* **Function parameter type**.
+It can be nullable, but it cannot be omitted. If you have that parameter and want it to be `null` in Java, you must send an `undefined` there during the function call.
```
Comment (haijian-vaadin) [inline articles/fusion/advanced/endpoints-generator.asciidoc:101]: one sentence per line, i.e. put If you ... in a new line.

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 2]
```diff
@@ -92,35 +93,66 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
...
+export async function getConnections(): Promise<Record<string, string> | undefined> {
+  return client.call('PersonEndpoint', 'getConnections');
 }
 ----
 
-In this case, if user calls `UserEndpoints.setName('first', 'last')` in TypeScript, the middle name is omitted and set as `null` when receiving in Java `UserEndpoints.setName` method.
-If the JavaType for that parameter is `Optional`, it receives `Optional.empty()`;
+==== Data Class Properties
+
+For the data class, there are one thing that can be nullable:
```
Comment (haijian-vaadin) [inline articles/fusion/advanced/endpoints-generator.asciidoc:150]: For the data class, one thing can be nullable:

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 3]
```diff
@@ -146,25 +178,31 @@ export default interface MyBean {
 }
 ----
 
-==== Return Type
+==== Collection Item Types
+
+For the collection types, there are one thing that can be nullable:
```
Comment (haijian-vaadin) [inline articles/fusion/advanced/endpoints-generator.asciidoc:183]: For the collection types, one thing can be nullable:

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 4]
```diff
@@ -92,35 +93,66 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
 * `org.eclipse.jdt.annotation.NonNull`
 * any other annotation (including custom) that has the `nonnull` (case-insensitive) name.
 
-.Required parameters in Java endpoint
+==== Endpoint Functions
+
+For the endpoint function, there are two things that can be nullable:
```
Comment (haijian-vaadin) [inline articles/fusion/advanced/endpoints-generator.asciidoc:98]: two things can be nullable

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]

**Review by haijian-vaadin — APPROVED** [review]

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 5]
```diff
@@ -92,35 +92,67 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
...
+export async function getConnections(): Promise<Record<string, string> | undefined> {
+  return client.call('PersonEndpoint', 'getConnections');
 }
 ----
 
-In this case, if user calls `UserEndpoints.setName('first', 'last')` in TypeScript, the middle name is omitted and set as `null` when receiving in Java `UserEndpoints.setName` method.
-If the JavaType for that parameter is `Optional`, it receives `Optional.empty()`;
+==== Data Class Properties
+
+For the data class, one thing can be nullable:
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:150]: For me this literal “one thing” mention and the list feel redundant. I’d suggest to remove the list, and only keep the paragraph that says that class properties are nullable.

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 6]
```diff
@@ -92,35 +92,67 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
 * `org.eclipse.jdt.annotation.NonNull`
 * any other annotation (including custom) that has the `nonnull` (case-insensitive) name.
 
-.Required parameters in Java endpoint
+==== Endpoint Functions
+
+For the endpoint function, two things can be nullable:
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:97]: “Nullable” implies the ability, “can be” feels redundant.

```suggestion
For the endpoint function, two things are nullable:
```

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 7]
```diff
@@ -92,35 +92,67 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
...
+
+* **Function parameter type**.
+It can be nullable, but it cannot be omitted.
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:100]: A bit mixing the parameters (the receiving variables) and the arguments (values that are being send) here, let’s clarify.

```suggestion
Arguments cannot be omitted even when the parameter types are nullable.
```

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 8]
```diff
@@ -92,35 +92,67 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
...
+If you have that parameter and want it to be `null` in Java, you must send an `undefined` there during the function call.
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:101]:
```suggestion
To receive a `null` parameter value in Java, send an `undefined` argument in the endpoint function call.
```

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 9]
```diff
@@ -146,25 +178,31 @@ export default interface MyBean {
 }
 ----
 
-==== Return Type
+==== Collection Item Types
+
+For the collection types, one thing can be nullable:
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:183]: Also redundant “one thing” list here.

**Review by platosha — CHANGES_REQUESTED** [review]

**Review by haijian-vaadin — APPROVED** [review]

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 10]
```diff
@@ -92,35 +92,67 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
...
 ----
 
-In this case, if user calls `UserEndpoints.setName('first', 'last')` in TypeScript, the middle name is omitted and set as `null` when receiving in Java `UserEndpoints.setName` method.
-If the JavaType for that parameter is `Optional`, it receives `Optional.empty()`;
+==== Data Class Properties
+
+For the data class, nullable elements are:
 
-==== Properties
+* **Properties**.
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:152]:
```suggestion
```

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 11]
```diff
@@ -92,35 +92,67 @@ You may use any annotation that has a `nonnull` (case-insensitive) name, for exa
...
+export async function getConnections(): Promise<Record<string, string> | undefined> {
+  return client.call('PersonEndpoint', 'getConnections');
 }
 ----
 
-In this case, if user calls `UserEndpoints.setName('first', 'last')` in TypeScript, the middle name is omitted and set as `null` when receiving in Java `UserEndpoints.setName` method.
-If the JavaType for that parameter is `Optional`, it receives `Optional.empty()`;
+==== Data Class Properties
+
+For the data class, nullable elements are:
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:150]: Looks redundant to have a list with just one item, I’d remove the list and describe the item right away.

```suggestion
Properties of the data classes are nullable.
```

File: articles/fusion/advanced/endpoints-generator.asciidoc [hunk 12]
```diff
@@ -146,25 +178,31 @@ export default interface MyBean {
 }
 ----
 
-==== Return Type
+==== Collection Item Types
 
-.Required return type in Java
+For collection types, nullable elements are:
+
+* **Item type**.
```
Comment (platosha) [inline articles/fusion/advanced/endpoints-generator.asciidoc:185]:
```suggestion
The collection item type is nullable.
```

**Review by platosha — APPROVED** [review]

**Comment by platosha** [comment]
Approved with some non-blocking suggestions. Feel free to skip them and merge as is.

**Review by platosha — APPROVED** [review]

**Pull request merged**